apps/heatsuite/heatsuite.app.js

```diff
@@ -0,0 +1,184 @@
+          //Core Temperature
+          found = true;
+          layout.msg.label = "Temp Found";
+          layout.render();
+          if (NRFFindDeviceTimeout) clearTimeout(NRFFindDeviceTimeout);
+          return Bangle.load('heatsuite.bletemp.js');
+        }
+      });
+    }
+    if (!found) {
+      modHS.log("Search Complete, No Devices Found");
+      queueNRFFindDeviceTimeout();
+    } else {
+      if (TaskScreenTimeout) clearTimeout(TaskScreenTimeout);
+      if (NRFFindDeviceTimeout) clearTimeout(NRFFindDeviceTimeout);
+    }
+  }, { timeout: 3000, active: true});
+}
+
+function taskButtonInterpretter(string) {
+  //turn off FindDeviceHandler whenever we navigate off task screen
+  let command = 'if (NRFFindDeviceTimeout){clearTimeout(NRFFindDeviceTimeout);}' + string;
+  return eval(command);
+}
+
+function queueTaskScreenTimeout() {
+  if (TaskScreenTimeout) clearTimeout(TaskScreenTimeout);
+  if (TaskScreenTimeout === undefined) {
+    TaskScreenTimeout = setTimeout(function () {
+      if (NRFFindDeviceTimeout) clearTimeout(NRFFindDeviceTimeout);
+      Bangle.load();
+    }, 180000);
+  }
+}
+
+function draw() {
+  g.clear();
+  g.reset();
+  if (studyTasks.length === 0) {
+    if(require("Storage").list().includes("heatsuite.survey.json")){ //likely just using for EMA survey
+      return Bangle.load('heatsuite.survey.js'); //go right to survey!
+    }
+    modHS.log('No Study Tasks loaded...');
+    layout = new Layout({
+      type: "v",
+      c: [
+        {
+          type: "txt",
+          font: "Vector:30",
+          label: "No Study Tasks Loaded.",
+          wrap: true,
+          fillx: 1,
+          filly: 1
+        }
+      ]
+    });
+    layout.render();
+    return;
+  }
+  let taskArr = appCache.taskQueue;
+  let taskID = [];
+  if (taskArr !== undefined) {
+    taskID = taskArr.filter(function (taskArr) {
+      return taskArr.id;
+    }).map(function (taskArr) {
+      return taskArr.id;
+    });
+  }
+  let layoutOut = { type: "v", c: [] };
+  let row = { type: "h", c: [] };
+  let rowCount = 2;
+  if( studyTasks.length > 4){
+    rowCount = 3; //so we can include up to 9 tasks on the screen at once
+  }
+  studyTasks.forEach(task => {
+    let btn = { type: "btn", fillx: 1, filly: 1 };
+    btn.id = task.id;
+    btn.src = eval(task.icon);
+    //callback on button press
+    if (task.cbBtn) {
+      btn.cb = l => taskButtonInterpretter(task.cbBtn);
+    }
+    //back color determination
+    btn.btnFaceCol = "#90EE90";
+    //a to do!!
+    if (taskID.includes(task.id)) {
+      btn.btnFaceCol = "#FFFF00";
+    }
+    //no bt paired
+    if (task.btPair === true) {
+      if (settings["bt_" + task.id + "_id"] === undefined || !settings["bt_" + task.id + "_id"]) {
+        //make it clickable so we can go to settings and pair something
+        btn.btnFaceCol = "#FF0000";
+        btn.cb = l => taskButtonInterpretter("Bangle.load('heatsuite.settings.js');");
```

Comment:
Fixes @thyttan: "This happened when clicking either of the two red buttons (top right)." - I was incorrectly navigating to the settings so that a device can be paired. 

```suggestion
        btn.cb = l => eval(require("Storage").read("heatsuite.settings.js"))(()=>load());
```

Reply:
Great! 

If you want, instead of sending `()=>load()` as the callback back function of the settings menu, you could do `()=>load("heatsuite.app.js")` to go back to the app.